sdk/src/fee.rs

```diff
@@ -0,0 +1,60 @@
+/// TODO
+use crate::native_token::sol_to_lamports;
+
+// TODO privitize members
+/// A fee and its associated compute unit cap
+#[derive(Debug, Default, Clone)]
+pub struct FeeBin {
+    /// maximum compute units for which this fee should be charged
+    pub cap: u64,
+    // fee in lamports
+    pub fee: u64,
+}
+
+// TODO privitize mrmbers
+/// Information used to calculate fees
+#[derive(Debug, Clone)]
+pub struct FeeStructure {
+    // lamports per signature
+    pub lamports_per_signature: u64,
+    // fee bins
+    pub bins: Vec<FeeBin>,
+}
+impl FeeStructure {
+    pub fn new(sol_per_signature: f64, bins: Vec<(u64, f64)>) -> Self {
+        let bins = bins
+            .iter()
+            .map(|(cap, sol)| FeeBin {
+                cap: *cap,
+                fee: sol_to_lamports(*sol),
+            })
+            .collect::<Vec<_>>();
+        FeeStructure {
+            lamports_per_signature: sol_to_lamports(sol_per_signature),
+            bins,
+        }
+    }
+
+    pub fn get_max_fee(&self, num_signatures: u64) -> u64 {
+        num_signatures
+            .saturating_mul(self.lamports_per_signature)
+            .saturating_add(self.bins.last().map(|bin| bin.fee).unwrap_or_default())
+    }
+}
+impl Default for FeeStructure {
```

Comment:
@aeyakovenko @sakridge 

The fee structure consists of a series of bins (cap, fee), when transactions come in they are charged based on which bin their requested compute units lands, and how many signatures their transaction contains.  If they don't request they get the highest bin.  The bins are currently a x10 lower fee for < 10k units, the current fee for 10k-200k units, and a doubling after that.  Can you provide feedback on these levels?  The activation of this feature will be via governance and discussed further beforehand but I'd like to get this initial shot in the ballpark.  I'm also working on a mechanism to which we can modify fee values at epoch boundaries without making a new release, the authority to do that will also be managed by a governance program.